internal/engine/compiler/engine.go

```diff
@@ -1065,12 +1074,27 @@ entry:
 			stack := ce.stack[base : base+stackLen]
 
 			fn := calleeHostFunction.parent.goFunc
-			switch fn := fn.(type) {
-			case api.GoModuleFunction:
-				fn.Call(ctx, ce.callerModuleInstance, stack)
-			case api.GoFunction:
-				fn.Call(ctx, stack)
-			}
+			func() {
+				defer func() {
```

Comment:
this defer causes allocation every time a Go host func is called. would it be possible to avoid when snapshot is disabled?